Sources/Extensions/MapKit/MKMapViewExtensions.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  MKMapViewExtensions.swift
+//  SwifterSwift
+//
+//  Created by Hannes Staffler on 24.01.19.
+//  Copyright © 2019 SwifterSwift
+//
+
+#if canImport(MapKit)
+import MapKit
+
+// MARK: - Initializers
+#if !os(watchOS)
+public extension MKMapView {
+
+    /// SwifterSwift: Register MKAnnotationView using class name
+    ///
+    /// - Parameter name: MKAnnotationView type
+    @available(iOS 11.0, *)
+    public func register<T: MKAnnotationView>(annotationViewWithClass name: T.Type) {
+        register(T.self, forAnnotationViewWithReuseIdentifier: String(describing: name))
+    }
+
+    /// SwifterSwift: Dequeue reusable MKAnnotationView using class name
+    ///
+    /// - Parameters:
+    ///   - name: MKAnnotationView type.
+    /// - Returns: optional MKAnnotationView object.
+    public func dequeueReusableAnnotationView<T: MKAnnotationView>(withClass name: T.Type) -> T? {
+        return dequeueReusableAnnotationView(withIdentifier: String(describing: name)) as? T
+    }
+
+    /// SwifterSwift: Dequeue reusable MKAnnotationView using class name
+    ///
+    /// - Parameters:
+    ///   - name: MKAnnotationView type.
+    ///   - annotation: annotation of the mapView.
+    /// - Returns: optional MKAnnotationView object.
+    @available(iOS 11.0, *)
+    public func dequeueReusableAnnotationView<T: MKAnnotationView>(withClass name: T.Type, for annotation: MKAnnotation) -> T? {
+        return dequeueReusableAnnotationView(withIdentifier: String(describing: name), for: annotation) as? T
```

Comment:
Should we trap here instead of returning an option like #439? 
The one above is fine because `dequeueReusableAnnotationView(withIdentifier: )` returns optional already. 
But this one maybe could be an option :))